src/poetry/packages/locker.py

```diff
@@ -587,12 +587,12 @@ def _dump_package(self, package: Package) -> dict[str, Any]:
 
         if package.extras:
             extras = {}
-            for name, deps in package.extras.items():
+            for name, deps in sorted(package.extras.items()):
                 # TODO: This should use dep.to_pep_508() once this is fixed
                 # https://github.com/python-poetry/poetry-core/pull/102
                 extras[name] = [
-                    dep.base_pep_508_name if not dep.constraint.is_any() else dep.name
-                    for dep in deps
```

Comment:
simply
```patch
-                extras[name] = [
+                extras[name] = sorted(
                     dep.base_pep_508_name if not dep.constraint.is_any() else dep.name
                     for dep in deps
-                ]
+                )
```
surely?

Reply:
Indeed, thanks. Amended in https://github.com/python-poetry/poetry/pull/6169/commits/ecbd171fe4984fd1b800503314d81be9b4c277be.